src/Imath/ImathMatrix.h

```diff
@@ -38,18 +38,63 @@ enum IMATH_EXPORT_ENUM Uninitialized
     UNINITIALIZED
 };
 
+namespace detail
+{
+template <typename T, size_t D> struct MatStorageAdapter
+{
+    struct data
+    {
+        T x[D][D];
+    };
+    using type = data;
+};
```

Comment:
If we're gonna mess with this anyway, should we make it [R][C] and leave room for future expansion to non-square matrices?

Reply:
shock and horror (probably)